### ISSUE TITLE
Special Duration Being Added to Pre-Defined Condition by addEffectWith

### ISSUE THREAD

**Issue by Jeznar** [DFreds/dfreds-convenient-effects#161]
**Describe the bug**
When I call **game.dfreds.effectInterface.addEffectWith()** to set a one-time SpecialDuration entry, the change I make is recorded in the running copy of the customized effect data so that doing the same thing multiple times stacks the change. The stack appears in the running copy and on the changed actors.  It is removed by a FoundryVTT reload from the CE data (as shown by a sparkling hand icon click) but remains on any modified tokens.

**To Reproduce**
I have created a little macro that can be run with one or more tokens selected.  I'll include that below.

1.  Make sure a customized _Stunned_ condition exists and make a note of any special conditions defined
2.  Create a copy of my **Convenient_Effects_Exercise.js** as a script macro (from below)
3. Select one or more target tokens (the first will be the origin)
4. Run Convenient_Effects_Exercise one or more times
5. Take a peak at the Stunned condition in CE's defined effect, checking out Special Duration field 

**Expected Behavior**

The Special duration change should appear only on the targeted tokens, not be reflected in to CE's copy of the data.

**Screenshots**
My predefined _Stunned_ effects in CE's data.

<img width="307" alt="1 Convenient_Effects_Stunned" src="https://user-images.githubusercontent.com/95548360/181913695-481ff431-e288-4e1a-a84c-914ec20ce631.png">

Duration tab of predefined custom stunned effect before my test.

<img width="906" alt="2 Stunned_as_initially_defined" src="https://user-images.githubusercontent.com/95548360/181913756-5cdda0f6-d6a8-4879-b4bf-d9e5128a584c.png">

Duration tab of predefined custom stunned after three executions of my test. Interestingly the name field has not been altered while the Special Duration has been.

<img width="904" alt="3 Stunned_after_three_executions" src="https://user-images.githubusercontent.com/95548360/181913779-ac917a6e-3996-4d35-9b29-b8c7391e036b.png">

Duration tab of one of my afflicted actors.  This one shows updated name and stacked up duration field from the forth execution.

<img width="906" alt="4 Stunned_on_Actor_after_four_executions" src="https://user-images.githubusercontent.com/95548360/181913813-a6a0ebd7-e368-41ae-a430-4c762f8129b3.png">

**Additional context**
Below is my testing macro to tickle my issue.

~~~javascript
const MACRO = "Convenient_Effects_Exercise.js"
const COND_APPLIED = "Stunned"  // A pre-existing custom effect managed by CE
const SPECDUR = "turnEndSource" // Special Duration that I want applied this one time

// Need to have selected at least one token
let aToken = canvas.tokens.controlled[0]
// ---------------------------------------------------------------------------------------
// Obtain and modify condition to be applied
//
let effect = game.dfreds.effectInterface.findEffectByName(COND_APPLIED).convertToObject();
effect.name = `${COND_APPLIED} ${SPECDUR}`;     // Change the effect's name
effect.flags.dae.specialDuration.push(SPECDUR)  // Add to the effect's special duration
// ---------------------------------------------------------------------------------------
// Loop through all of the selected tokens and apply the condition
//
for ( let tToken of canvas.tokens.controlled ) 
    game.dfreds.effectInterface.addEffectWith({effectData:effect, 
        uuid:tToken.actor.uuid, origin:aToken.actor.uuid });
~~~

**Comment by DFreds** [comment]
Yeah, this is definitely a bug. Thanks for bringing this to my attention as well as for the detailed explanation. I'll see what is going on

**Comment by DFreds** [comment]
I'm going to guess it's because of some conflict with using the `findEffectByName` before changing it but I'll see.

**Comment by DFreds** [comment]
I think I fixed this by using `deepClone` on the `convertToObject` method. I'll release this in v2.7.3 and we can reopen it if the problem persists.

**Comment by DFreds** [comment]
Fixed in v2.7.3

**Comment by Jeznar** [comment]
Fix confirmed!   My test case works after installing v2.7.3.  Thank you very much.
